crates/task-impls/src/transactions.rs

```diff
@@ -126,15 +126,16 @@ impl<
                     // send the sequenced transactions to VID and DA tasks
                     let block_view = if make_block { view } else { view + 1 };
                     let encoded_transactions = match block_data.block_payload.encode() {
-                        Ok(encoded) => encoded.into_iter().collect::<Vec<u8>>(),
+                        Ok(encoded) => encoded,
                         Err(e) => {
                             error!("Failed to encode the block payload: {:?}.", e);
                             return None;
                         }
                     };
+
                     broadcast_event(
                         Arc::new(HotShotEvent::BlockRecv(
-                            encoded_transactions,
+                            encoded_transactions.as_ref().to_vec(),
```

Comment:
ugh. I suspect the compiler is smart enough not to copy the entire payload here, since `block_data` goes out of scope in this block anyway. Still this is bad practice.

@bfish713 lots of events seem to demand an owned block payload. Is that necessary? If so then we should be passing ownership around instead of copying (eg. via `to_vec()`) whenever possible.

I wonder whether we should make a `BlockPayload` method that consumes self and returns the owned payload (and other stuff, such as the namespace table) for use cases such as this. 🤔 

Or... I suspect most events could just take ownership of the `BlockPayload` directly instead of extracting it's binary payload. It's certainly true for this `BlockRecv` event.